modules/documents/app/controllers/documents_controller.rb

```diff
@@ -124,14 +116,4 @@ def add_attachment
   def document_params
     params.fetch(:document, {}).permit('category_id', 'title', 'description')
   end
-
-  def notify_attachments(added)
-    return if added.empty?
-
-    users = added.first.container.recipients
-    users.each do |user|
-      UserMailer.attachments_added(user, added).deliver_later
```

Comment:
That mail is not sent out any more. 

We could extend the [DocumentStrategy](https://github.com/opf/openproject/blob/dev/modules/documents/app/services/notifications/mail_service/document_strategy.rb#L32) to check for an attachment having been added to the journal and then sending out the `attachments_added` mail.

Reply:
I don't think that will work. This method depended on calculating which attachments were newly added. Information we don't have later on as far as I can tell. 

Since there is no setting for "attachments added to documents", I believed this could be removed. I can re-add it to the document controller sending it to all users in the project that have the global notification `documents_added` though.

Reply:
FYI That's what I did. The documents controller now implements part of the strategies (checking allowed scope, recipients) for the attachments case. I also added some specs to ensure the emission of the mail is tested.